ledger/archival_test.go

```diff
@@ -678,11 +678,6 @@ func makeSignedTxnInBlock(tx transactions.Transaction) transactions.SignedTxnInB
 
 func TestArchivalFromNonArchival(t *testing.T) {
 	// Start in non-archival mode, add 2K blocks, restart in archival mode ensure only genesis block is there
-	deadlockDisable := deadlock.Opts.Disable
-	deadlock.Opts.Disable = true
-	defer func() {
-		deadlock.Opts.Disable = deadlockDisable
-	}()
```

Comment:
I guess it was disabled here to speed up the test

Reply:
no - it definitely wasn't placed here for speeding tests.
It was placed here since it would prevent random deadlocks on this test on travis.

 If you recall, we had quite many of these.. and this addressed the issue. Changing a global variable can lead to a data race, like in this case. The textbook solution I've used is to ensure each test run to completion ( i.e. shutting down all it's child threads ) before exiting.